test/integration/component/test_multiple_physical_network_creation.py

```diff
@@ -0,0 +1,410 @@
+
+        # Enable the network
+        self.physical_network_3.update(
+            self.apiclient,
+            tags="guest",
+            state="Enabled"
+        )
+
+        #2. try adding traffic type Guest
+        TrafficType.add(
+            self.apiclient,
+            physicalnetworkid=self.physical_network_3.id,
+            kvmnetworklabel=self.kvmnetworklabel,
+            traffictype="Guest"
+        )
+
+        # Create network offering
+        self.services["network_offering_shared"]["supportedservices"] = ""
+        self.services["network_offering_shared"]["serviceProviderList"] = {}
+        self.services["network_offering_shared"]["tags"] = "guest"
+        self.network_offering = NetworkOffering.create(
+            self.apiclient,
+            self.services["network_offering_shared"]
+        )
+
+        # Enable network offering
+        self.network_offering.update(
+            self.apiclient,
+            state='Enabled'
+        )
+
+        #3. Create a shared network
+        self.shared_network = Network.create(
+            self.apiclient,
+            self.services["network2"],
+            networkofferingid=self.network_offering.id,
+            zoneid=self.zone.id,
+            domainid=self.domain.id
+            #physicalnetworkid=self.physical_network_3.id
+        )
+        self.cleanup.append(self.shared_network)
+
+        # Create small service offering
+        self.service_offering = ServiceOffering.create(
+            self.apiclient,
+            self.testdata["service_offerings"]["small"]
+        )
+        self.cleanup.append(self.service_offering)
+
+        #4. Create virtual machine
+        self.testdata["virtual_machine"]["zoneid"] = self.zone.id
+        self.testdata["virtual_machine"]["template"] = self.template.id
+        self.virtual_machine = VirtualMachine.create(
+            self.apiclient,
+            self.testdata["virtual_machine"],
+            templateid=self.template.id,
+            serviceofferingid=self.service_offering.id,
+            networkids=self.shared_network.id
+        )
+        self.cleanup.append(self.virtual_machine)
+
+        list_vms = VirtualMachine.list(
+            self.apiclient,
+            id=self.virtual_machine.id
+        )
+
+        vm = list_vms[0]
+        # make sure that vm uses the same vlan of the network
+        self.assertEqual(vm.nic[0].traffictype,
+                         "Guest",
+                         "Vm traffic type is not same")
+
+        self.assertEqual(vm.nic[0].isolationuri,
+                         "vlan://" + str(self.services["network2"]["vlan"]),
+                         "Vm network vlan is not same")
+
+        self.cleanup.append(self.virtual_machine)
+        self.cleanup.append(self.service_offering)
+        self.cleanup.append(self.shared_network)
+        self.cleanup.append(self.network_offering)
+        self.cleanup.append(self.physical_network_3)
```

Comment:
@soreana these should be move up each just below the respective create API call they apply to. If not and expcption will prevent these from being cleaned.

Reply:
sorry, I see a few already are and those should be removed here completely.

Reply:
```suggestion
```

Reply:
I've updated the test, apparently I forgot to remove these :-)